src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/OsxIPGlobalProperties.cs

```diff
@@ -8,54 +8,24 @@ namespace System.Net.NetworkInformation
 {
     internal class OsxIPGlobalProperties : UnixIPGlobalProperties
     {
-        public override string DhcpScopeName
-        {
-            get
-            {
-                throw new PlatformNotSupportedException();
-            }
-        }
+        public override string DhcpScopeName { get { throw new PlatformNotSupportedException(); } }
 
-        public override string DomainName
-        {
-            get
-            {
-                return HostInformation.DomainName;
-            }
-        }
+        public override string DomainName { get { return HostInformation.DomainName; } }
 
-        public override string HostName
-        {
-            get
-            {
-                return HostInformation.HostName;
-            }
-        }
+        public override string HostName { get { return HostInformation.HostName; } }
 
-        public override bool IsWinsProxy
-        {
-            get
-            {
-                return false;
-            }
-        }
+        public override bool IsWinsProxy { get { throw new PlatformNotSupportedException(); } }
 
-        public override NetBiosNodeType NodeType
-        {
-            get
-            {
-                return NetBiosNodeType.Unknown;
-            }
-        }
+        public override NetBiosNodeType NodeType { get { return NetBiosNodeType.Unknown; } }
 
         public unsafe override TcpConnectionInformation[] GetActiveTcpConnections()
         {
             int realCount = Interop.Sys.GetEstimatedTcpConnectionCount();
-            int infoCount = (int)(realCount * 1.5f);
+            int infoCount = realCount * 2;
             Interop.Sys.NativeTcpConnectionInformation* infos = stackalloc Interop.Sys.NativeTcpConnectionInformation[infoCount];
             while (Interop.Sys.GetActiveTcpConnectionInfos(infos, &infoCount) == -1)
             {
-                var newAlloc = stackalloc Interop.Sys.NativeTcpConnectionInformation[infoCount];
+                Interop.Sys.NativeTcpConnectionInformation* newAlloc = stackalloc Interop.Sys.NativeTcpConnectionInformation[infoCount];
```

Comment:
Yeah, there is the potential. I'm actually considering not re-trying at all here. Most of the example code I have seen just tries to call the function once, with a buffer of the estimated size. If that call fails, I have just seen people give up. Additionally, I don't think I have actually seen the first call to this fail, either in here or in the native shim which does a similar thing. The first call here, to "GetEstimatedTcpConnectionCount" is supposed to give you a size that will be good enough if you call again in a short amount of time.